datadog_checks_base/datadog_checks/base/checks/win/wmi/base.py

```diff
@@ -303,6 +311,40 @@ def _get_wmi_properties(self, instance_key, metrics, tag_queries):
 
         return self._wmi_props
 
+    def validate_tag_queries_aliases(self, tag_queries):
+        # type: (List[TagQuery]) -> None
+        """
+        Validate tag_queries configuration to ensure aliases are provided when 'AS' is used.
+        Logs warnings for any invalid configurations.
+        """
+        for tag_query in tag_queries:
+            if len(tag_query) < 4:
+                continue
+            target_property_str = tag_query[3]
+            property, alias = self.parse_alias(target_property_str)
+            # Check if 'AS' was present but alias is empty
+            if (' AS' in target_property_str or ' as' in target_property_str) and alias is None:
+                self.log.warning(
+                    "No alias provided after 'AS' for property: %s in tag_queries. Using property for tag", property
+                )
```

Comment:
Looks like this can be moved into the parse_alias function here and as a warning there https://github.com/DataDog/integrations-core/pull/21792/files#diff-bf9849fd6e2bb36683eb8a1f52d6f6093da13c4fd5e1e5822e5e00eddc6bb07aR40